ovh/resource_ip_move.go

```diff
@@ -0,0 +1,309 @@
+			endpoint := fmt.Sprintf("/ip/%s/move",
+				url.PathEscape(ip),
+			)
+			if err := config.OVHClient.Post(endpoint, opts, ipTask); err != nil {
+				return fmt.Errorf("calling Post %s: %q", endpoint, err)
+			}
+		}
+		d.SetId(fmt.Sprint(ipTask.TaskId))
+		if err = d.Set("task_start_date", ipTask.StartDate.String()); err != nil {
+			return err
+		}
+
+		_, err = waitForTaskFinished(d, meta, ipTask, ip, opts)
+		if err != nil {
+			return err
+		}
+	}
+	err = resourceIpRead(d, meta)
+	return err
+}
+
+// waitForTaskFinished queries GET /ip/:ip/task/:taskId route until task state is in a terminal success or error state or until WaitingTimeInSecondsBeforeRefreshState is reached
+// and returns :
+//   - finishedWithSuccess : true if task ended with success, false if ended with error, nil if not ended at all
+//   - err : nil if no error is encountered in the process, any met error otherwise
+//
+// in any case before returning, "task_status" field of d will be updated with the last known ipTask.Status
+func waitForTaskFinished(d *schema.ResourceData, meta interface{}, ipTask *IpTask, ip string, opts *IpMoveOpts) (finishedWithSuccess *bool, err error) {
+	defer func() {
+		errSet := d.Set("task_status", ipTask.Status)
+		if errSet == nil {
+			err = errSet
+		}
+	}()
+	return recursiveWaitTaskFinished(d, meta, ipTask, ip, opts)
+}
+
+// recursiveWaitTaskFinished checks a given ipTask and return true if task status is in a state that we consider finished.
+// and calls itself again if task is not finished while task is not yet expired
+func recursiveWaitTaskFinished(d *schema.ResourceData, meta interface{}, ipTask *IpTask, ip string, opts *IpMoveOpts) (finished *bool, err error) {
+	switch ipTask.Status {
+	case IpTaskStatusDone:
+		return helpers.GetNilBoolPointer(true), nil
+	case IpTaskStatusCancelled, IpTaskStatusOvhError, IpTaskStatusCustomerError:
+		return helpers.GetNilBoolPointer(true), fmt.Errorf("could not assign IP %s to service %v as Ip task is %s", ip, opts.To, ipTask.Status)
+	}
+	timeDiff := time.Now().Sub(ipTask.StartDate)
+	if timeDiff < taskExpiresAfter {
+		log.Printf("[DEBUG] ipTask.Status is currently: %s. Waiting %d second (we allow %f more seconds for the task to complete)", ipTask.Status, WaitingTimeInSecondsBeforeRefreshState, taskExpiresAfter.Seconds()-timeDiff.Seconds())
+		time.Sleep(WaitingTimeInSecondsBeforeRefreshState * time.Second)
+		err = resourceIpTaskRead(d, ipTask, meta)
+		return recursiveWaitTaskFinished(d, meta, ipTask, ip, opts)
+	}
+	log.Printf("[WARNING] - waitForTaskFinished max number of retries reached without the task having reached a terminal state")
+	return nil, nil
+}
+
+func resourceIpTaskRead(d *schema.ResourceData, ipTask *IpTask, meta interface{}) error {
+	config := meta.(*Config)
+
+	endpoint := fmt.Sprintf("/ip/%s/task/%d",
+		url.PathEscape(d.Get("ip").(string)),
+		ipTask.TaskId,
```

Comment:
needs to be `url.PathEscape`'d

Reply:
but ipTask.TaskId is an int64, not a string (see %d formatter)

Reply:
my bad, read too fast :)